### ISSUE TITLE
Accesories removed from home app after upgrade to 8.0

### ISSUE THREAD

**Issue by arkcolson** [ztalbot2000/homebridge-cmd4#147]
Hi

After the upgrade to 8.0 I noticed that the accesories were removed from the home app. The logs are saying:

[8/13/2024, 10:19:50 AM] [Cmd4] Cmd4Platform didFinishLaunching
[8/13/2024, 10:19:50 AM] [Cmd4] Restoring existing accessory from cache: TV Joppe
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Ventilatie
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Laadstation
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Capaciteitstarief
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Zonnepanelen
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Heat L 50%
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Heat L 100%
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Heat A 50%
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Heat A 100%
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Heat R 50%
[8/13/2024, 10:19:55 AM] [Cmd4] Restoring existing accessory from cache: Heat R 100%
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory TV Joppe will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Ventilatie will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Laadstation will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Capaciteitstarief will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Zonnepanelen will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Heat L 50% will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Heat L 100% will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Heat A 50% will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Heat A 100% will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Heat R 50% will be removed.
[8/13/2024, 10:19:55 AM] [Cmd4] Accessory Heat R 100% will be removed.

After reverting to version 7.1 the accesories were restored:
[8/13/2024, 11:28:41 AM] [Cmd4] Cmd4Platform didFinishLaunching
[8/13/2024, 11:28:41 AM] [Cmd4] Adding new platformAccessory: TV Joppe
[8/13/2024, 11:28:41 AM] [Cmd4] Configuring platformAccessory: TV Joppe
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Ventilatie
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Ventilatie
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Laadstation
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Laadstation
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Capaciteitstarief
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Capaciteitstarief
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Zonnepanelen
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Zonnepanelen
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Heat L 50%
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Heat L 50%
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Heat L 100%
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Heat L 100%
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Heat A 50%
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Heat A 50%
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Heat A 100%
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Heat A 100%
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Heat R 50%
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Heat R 50%
[8/13/2024, 11:28:45 AM] [Cmd4] Adding new platformAccessory: Heat R 100%
[8/13/2024, 11:28:45 AM] [Cmd4] Configuring platformAccessory: Heat R 100%

Any idea why this could happen?

Br
Arkadi

**Comment by ztalbot2000** [comment]
I think you will find that this is a homebridge 2,0 behavior for them to update their accessories to a new format. If you run Cmd4 a second time, this message does not appear. The same thing will happen if you downgrade to 1.8.
Give it a try to see what I mean.
John

**Comment by pixeleyesltd** [comment]
I experienced this issue too. The NodeJS version required by this plugin (20.15.1) appears to be later than the NodeJS version shipped in the current official HomeBridge Docker (20.15.0), so the plugin failed to launch and accessories were removed. Unless there's a reason for it, perhaps lowering the NodeJS requirement to 20.15.0 would be a good thing for the users of this plugin.

**Comment by ztalbot2000** [comment]
Hi,

  Thanks for the information.  The Cmd4 package.json says "^18.20.4 ||
^20.15.1". I'd be kind of surprised if 20.15.0 to 20.15.1 would be the
difference, but I will check it out. I did see the behaviour you're
mentioning, but it disappeared when I ran homebridge a second time with the
exact same configuration. It could not hurt for me to check out 20.15.0
though.

ttyl,
John

**Comment by ztalbot2000** [comment]
Hi,

  So homebridge at
https://github.com/homebridge/homebridge/wiki/Updating-To-Homebridge-v2.0
says node should be defined as:
```text

"engines": {
    "homebridge": "^1.6.0 || ^2.0.0-beta.0",
    "node": "^18.20.4 || ^20.15.1"
  },

```
So the package.json is correct. There is still a problem with accessories
appearing to be deleted. Starting Cmd4 a second time and this disappears. I
just found out that starting it a third time and it happens again. This is
really a homebridge bug and I will verify that today and upon doing so will
create a bug report on their side.
ttyl,
John

**Comment by ztalbot2000** [comment]
Closed with homebridge-cmd4 v8.0.1
It was no longer necessary to remove old cached accessories that are not in your config.json as they never get re-allocated. Unfortunately this made it look like current accessories were added and removed.

Thanks for enjoying Cmd4
John Talbot
ztalbot2000@gmail.com

**Comment by ztalbot2000** [comment]
closed with last comment
